src/hooks/useLocalStorageAuth.tsx

```diff
@@ -0,0 +1,27 @@
+import { useEffect, useState } from 'react';
+
+import { LocalStorageKey, setLocalStorageValue, getLocalStorageValue, removeLocalStorageValue } from 'lib/localStorage';
+import { Auths } from 'types/auth';
+
+type Nullable<T> = T | null;
+
+const useLocalStorageAuth = (): [Nullable<Auths>, (authData: Nullable<Auths>) => void] => {
+  const [auth, setAuth] = useState<Nullable<Auths>>(null);
+
+  useEffect(() => {
+    const authData = getLocalStorageValue(LocalStorageKey.auth);
+    setAuth(authData);
+  }, []);
+
+  useEffect(() => {
+    if (auth) {
+      setLocalStorageValue(LocalStorageKey.auth, auth);
+    } else {
```

Comment:
Do we need this condition? It seems like we will set the same thing that we get from the above `useEffect` 🤔 
`const authData = getLocalStorageValue(LocalStorageKey.auth); setAuth(authData);` then set the auth back to the local storage

Reply:
With that, if we refresh the page after login it will go to the login page because` authData` is null and there's nothing to set to the localStorage.
The first one is to check the local storage after refresh or reopen the page with an unexpired token and the second is use to set auth data after login or remove it after logout